File: php/examples/streaming.php
<?php

declare(strict_types=1);

require __DIR__ . '/../tts/Client.php';

use Voxygen\TTS\Client;

// TTS request parameters
$parameters = [
    "voice" => "Mary", // Select a voice available on your subscription
    "language" => "en-US",
    "text" => "Hello. How can I help you today ?", // Text to synthesize
    "header" => "headerless",
];

$client = new Client(token: "YOUR_TOKEN");

$query = $client->buildRequest($parameters);
$stream = $client->open($query);
$headers = Client::response_headers($stream);

// Read and report audio chunks
$chunkSize = 4096;
while (!feof($stream)) {
    $data = fread($stream, $chunkSize);
    if ($data === false || $data === '') {
        break;
    }
    echo "received chunk of audio data: " . strlen($data) . " bytes" . PHP_EOL;
}

Client::close($stream);

File: python/examples/streaming.py
from voxygen.tts import Client

# TTS request parameters
parameters = {
    "voice": "Mary",  # Select a voice available on your subscription
    "language": "en-US",
    "text": "Hello. How can I help you today ?",  # Text to synthesize
    "header": "headerless",
}

with Client(token="YOUR_TOKEN") as client:
    request = client.build_request(parameters)
    response = client.send(request)

    if client.content_type(response) == Client.MimeType.AUDIO:
        for data in client.iter_data(response):
            print(f"received chunk of audio data: {len(data)} bytes")
